### ISSUE TITLE
Implement journal field

### ISSUE THREAD

**Issue by alejandromumo** [inveniosoftware/invenio-rdm-records#1191]
Implement the field `journal` considering its nested attributes:

- `journal.pages`
- `journal.title`
- `journal.volume`
- `journal.year`

Take into account the following requirements when implementing this field:

- It is needed in zenodo-rdm. 
- It must be a custom field. 
- It must be searchable. Search must be compatible with the current [Zenodo search guide](https://help.zenodo.org/guides/search/)
- If needed, update DataCite serializer taking into account [Datacite Schema 4.4 docs](https://support.datacite.org/docs/datacite-metadata-schema-44). 
  - A recommended field [`RelatedItem` ](https://support.datacite.org/docs/datacite-metadata-schema-v44-recommended-and-optional-properties#20-relateditem) can be used to serialize journal's information. 
- The upload form should be similar to zenodo (e.g. `Journal` has its own section)

**Comment by alejandromumo** [comment]
Currently, nested fields are not supported natively if we want to re-use UI widgets, even though the backend seems to be working for now. 

After irl discussion, we decided to move forward with [creating a custom field](https://inveniordm.docs.cern.ch/develop/howtos/custom_fields/) (e.g. `JournalCF`) for each field and implement the react component + jinja template for every field (`journal`. `meeting`, etc). For now it seems to be the less risky option and it unblocks the issue

**Comment by alejandromumo** [comment]
The serializer to Datacite is not yet implemented.

@slint should it be implemented in this issue or we tackle it afterwards?

**Comment by slint** [comment]
Let's split into a separate issue, since there's also other work being done at this moment on the serializers. Also I don't remember exactly how we decided back in the day to allow serializing custom fields into the different export formats (there was some "plugin/extension" interface but I don't recall specifics). CC @ppanero @zzacharo can also chip-in

**Comment by alejandromumo** [comment]
> Let's split into a separate issue, since there's also other work being done at this moment on the serializers. Also I don't remember exactly how we decided back in the day to allow serializing custom fields into the different export formats (there was some "plugin/extension" interface but I don't recall specifics). CC @ppanero @zzacharo can also chip-in

Moreover, in terms of ES search we should guarantee that this custom field maps to `journal.*`, as it was in zenodo.

**Comment by slint** [comment]
> Moreover, in terms of ES search we should guarantee that this custom field maps to `journal.*`, as it was in zenodo. Perhaps using the search redirector is enough.
> 
> In case we want this straightaway I can implement it in this issue.

I think we should do this on the ZenodoRDM instance config directly. Although I could also see a "contrib" set of search redirects/transforms... In any case, since we need them you could start testing it out, and afterwards we figure out what makes sense.

**Comment by ppanero** [comment]
@slint export formats have not been implemented for custom fields. There were a few approaches but it was cut of due to time constraints. See [here](https://github.com/inveniosoftware/rfcs/blob/master/rfcs/rdm-0064-custom-fields.md#export-formats) in the RFC.

**Comment by alejandromumo** [comment]
![Screenshot at Mar 01 17-30-32](https://user-images.githubusercontent.com/21204744/222202306-1645b851-b128-4c8c-9df9-d0b271f9e4ba.png)
![Screenshot at Mar 01 17-31-55](https://user-images.githubusercontent.com/21204744/222202414-e83294d6-2587-4ace-94f5-68f1a9d630e1.png)
